permissions_validators/src/lib.rs

```diff
@@ -98,6 +98,65 @@ macro_rules! try_into_or_exit {
     };
 }
 
+macro_rules! declare_token {
+    (
+        $(#[$outer_meta:meta])* // Structure attributes
+        $ident:ident {          // Structure definiton
+            $(
+                $param_name:ident ($param_string:literal): $param_typ:ty
+             ),* $(,)? // allow trailing comma
+        },
+        $string:tt // Token name
+    ) => {
+
+        $(#[$outer_meta])*
+        ///
+        /// A wrapper around [PermissionToken](iroha_data_model::permissions::PermissionToken).
+        #[derive(iroha_schema::IntoSchema)]
+        pub struct $ident
+        where $($param_typ: Into<Value>,)* {
+            $(
+                $param_name : $param_typ
+             ),*
+        }
+
+        impl $ident {
+            /// Get associated [`PermissionToken`](iroha_data_model::permissions::PermissionToken) name.
+            pub fn name() -> &'static Name {
```

Comment:
I don't think we should have this method because we're relying on nominal types here. If it's possible to make it work with generic tokens.

Reply:
I'm not sure I follow.
Right now it's used in comparisons like [this](https://github.com/hyperledger/iroha/pull/2156/files#diff-9fe2feb360cc0bdcb41fc22b6a8a2cabfdbff7bb7cecafd9bd3747202a37f1f9R112) instead of previously defined consts. How would you propose getting rid of it?

Reply:
you can do .into() and then compare. I think I'd like to get rid of the name method on specialized tokens